File: src/main/java/com/atsuishio/superbwarfare/init/ModPotions.java
package com.atsuishio.superbwarfare.init;

import net.minecraft.world.effect.MobEffectInstance;
import net.minecraft.world.item.alchemy.Potion;

import java.util.function.Supplier;

public class ModPotions {

    public static final Supplier<Potion> SHOCK = Registration.potion("superbwarfare_shock",
            () -> new Potion(new MobEffectInstance(ModMobEffects.SHOCK.get(), 100, 0)));
    public static final Supplier<Potion> STRONG_SHOCK = Registration.potion("superbwarfare_strong_shock",
            () -> new Potion(new MobEffectInstance(ModMobEffects.SHOCK.get(), 100, 1)));
    public static final Supplier<Potion> LONG_SHOCK = Registration.potion("superbwarfare_long_shock",
            () -> new Potion(new MobEffectInstance(ModMobEffects.SHOCK.get(), 400, 0)));
}

File: src/main/kotlin/com/atsuishio/superbwarfare/init/ModPotions.kt
package com.atsuishio.superbwarfare.init

import com.atsuishio.superbwarfare.Mod
import net.minecraft.core.Registry
import net.minecraft.core.registries.BuiltInRegistries
import net.minecraft.world.effect.MobEffectInstance
import net.minecraft.world.item.alchemy.Potion

@Suppress("unused")
object ModPotions {
    @JvmField
    val SHOCK: Potion = register(
        "superbwarfare_shock",
        Potion(MobEffectInstance(ModMobEffects.SHOCK, 100, 0))
    )

    @JvmField
    val STRONG_SHOCK: Potion = register(
        "superbwarfare_strong_shock",
        Potion(MobEffectInstance(ModMobEffects.SHOCK, 100, 1))
    )

    @JvmField
    val LONG_SHOCK: Potion = register(
        "superbwarfare_long_shock",
        Potion(MobEffectInstance(ModMobEffects.SHOCK, 400, 0))
    )

    private fun register(name: String, potion: Potion): Potion {
        return Registry.register(BuiltInRegistries.POTION, Mod.loc(name), potion)
    }

    @JvmStatic
    fun init() {
    }
}
